source/features/github-actions-indicators.tsx

```diff
@@ -25,7 +46,12 @@ function addTooltip(element: HTMLElement, tooltip: string): void {
 	}
 }
 
-const getWorkflowsDetails = cache.function('workflows', async (): Promise<Record<string, WorkflowDetails> | false> => {
+const getWorkflows = async (): Promise<Workflow[]> => {
+	const response = await api.v3(`/repos/${getRepo()!.owner}/${getRepo()!.name}/actions/workflows`);
```

Comment:
I've tried to resolve with v4 API, But I did not figured out to get Workflows.

I will try again to figure out. Thanks for yout notes!

Reply:
I'm looking into it too: https://docs.github.com/en/graphql/reference/objects#workflow

I thought it would be this, but it doesn't work:

```
{
  repository(owner: "refined-github", name: "refined-github") {
    workflowFiles: object(expression: "HEAD:.github/workflows") {
      ... on Tree {
        entries {
          name
          object {
            ... on Workflow {
              id
              state
              url
              createdAt
              updatedAt
            }
            ... on Blob {
              text
            }
          }
        }
      }
    }
  }
}
```

You can test it on https://docs.github.com/en/graphql/overview/explorer

Reply:
Yeah.. the v4 API is not fluent on Workflows from Repository.

Reply:
It seems that [gh-cli](https://cli.github.com/manual/gh_workflow_list) also uses v3 for this

- https://github.com/cli/cli/blob/09e86d0ebf5d8afd0c49770c5a580ee04c039c00/pkg/cmd/workflow/shared/shared.go#L61

I think you can only get to `Workflow` through a `WorkflowRun` on a commit, which is not reliable for us.

- https://stackoverflow.com/a/74868904/288906

Reply:
May the resources query is useful

Reply:
```graphql
{
  resource(
    url: "https://github.com/refined-github/refined-github/actions/workflows/labeler.yml"
  ) {
    url
    resourcePath
    ... on Workflow {
      name
    }
  }
}
```

We can get the Workflow with this query.
But there is no way to get "Workflow List". So we should use v3 api.